### DISCUSSION TITLE
Question about cache strategies

### DISCUSSION THREAD

**Discussion by vmgbrown** [infinispan/infinispan#11667] [Q&A] [answered]
The application I am working on uses Hibernate 5.4.18 and Infinispan 12.1.7. We have a number of entities annotated with `@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)`. The cache container for these entities is configured as follows:

```xml
<cache-container name="VmapUnmanaged" marshaller="JBOSS" modules="org.infinispan.hibernate-cache">
    <transport lock-timeout="60000"/>
    <local-cache name="local-query">
        <heap-memory size="10000"/>
        <expiration max-idle="100000"/>
    </local-cache>
    <invalidation-cache name="entity" remote-timeout="120000">
        <heap-memory size="10000"/>
        <expiration max-idle="120000" lifespan="600000"/>
    </invalidation-cache>
    <replicated-cache name="timestamps" remote-timeout="120000">
        <expiration interval="0"/>
    </replicated-cache>
</cache-container>
```

As mentioned in some earlier threads, we have observed some unexpected behavior related to these caches. Application performance degrades over time to the point where it is consistent with an empty or disabled L2 cache. We are still trying to identify the cause of this issue.

The [documentation](https://infinispan.org/docs/stable/titles/integrating/integrating.html#cache_strategies) says this:

> Support for replicated/distributed caches for read-write and read-only strategies...requires exclusively non-transactional configuration

It also says this:

> With Hibernate 5.3 the support for transactional caches has been dropped completely, and both read-write and transactional use the same implementation

Since we are using Hibernate 5.4, is it safe to assume that we are using "exclusively non-transactional configuration", or is that something that must be explicitly defined? If so, where?

We are performing evictions on these caches, which I know is not supported for distributed or replicated caches. However, it is supported for invalidation caches in non-transactional mode. How can I confirm that our caches are non-transactional?

Thank you,
Greg

**Comment by vmgbrown** [comment]
For what it's worth, the previous version of our application used WildFly 20, Hibernate 5.4.18, and Infinispan 10.1.8. I would be interested to know if there were any changes between 10.1.8 and 12.1.7 that might contribute to the behavior described above:

>Application performance degrades over time to the point where it is consistent with an empty or disabled L2 cache

**Comment by wburns** [comment, marked as answer]
> Since we are using Hibernate 5.4, is it safe to assume that we are using "exclusively non-transactional configuration", or is that something that must be explicitly defined? If so, where?

It is explicitly defined, which you are already doing in your cache configuration above. There is a transaction xml element that defines this for a given cache configuration. Since it is not present it will be non transactional.

Note that the infinispan-hibernate 2LC implementation has checks to verify this that look like https://github.com/infinispan/infinispan/blob/d297a88e1fc3a7f576f306595d8de087834afb97/hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/DataType.java#L30 so it should not be possible to start up with transactions in the first place.

> Application performance degrades over time to the point where it is consistent with an empty or disabled L2 cache. We are still trying to identify the cause of this issue.

The first place to start here is to check the stats to ensure you are getting hits as appropriate.
